src/Init/Data/UInt.lean

```diff
@@ -32,10 +32,10 @@ def UInt8.land (a b : UInt8) : UInt8 := ⟨Fin.land a.val b.val⟩
 def UInt8.lor (a b : UInt8) : UInt8 := ⟨Fin.lor a.val b.val⟩
 @[extern c inline "#1 ^ #2"]
 def UInt8.xor (a b : UInt8) : UInt8 := ⟨Fin.xor a.val b.val⟩
-@[extern c inline "#1 << #2"]
-def UInt8.shiftLeft (a b : UInt8) : UInt8 := ⟨a.val <<< b.val⟩
-@[extern c inline "#1 >> #2"]
-def UInt8.shiftRight (a b : UInt8) : UInt8 := ⟨a.val >>> b.val⟩
+@[extern c inline "#1 << #2 % 8"]
+def UInt8.shiftLeft (a b : UInt8) : UInt8 := ofNat (a.toNat <<< (b.toNat % 8))
```

Comment:
Thanks for finding this issue.  The modulus should be reduced to an and by C compilation optimizations, so this looks fine to me.
You could write `⟨a.val <<< UInt8.val (b % 8)⟩` to keep it at the Fin level rather than Nat, but I personally don't have much preference.

Reply:
> You could write `⟨a.val <<< UInt8.val (b % 8)⟩` to keep it at the Fin level rather than Nat

I agree, that would be slightly nicer